prometheus/labels.go

```diff
@@ -31,6 +31,64 @@ import (
 // create a Desc.
 type Labels map[string]string
 
+// ConstrainedLabels represents a label name and its constrain function
+// to normalize label values. This type is commonly used when constructing
+// metric vector Collectors.
+type ConstrainedLabel struct {
+	Name       string
+	Constraint func(string) string
+}
+
+func (cl ConstrainedLabel) Constrain(v string) string {
+	if cl.Constraint != nil {
+		return cl.Constraint(v)
+	}
+	return v
+}
+
+// ConstrainableLabels is the interface to create label constraints
```

Comment:
Let's also add your example of how to use it in the comment... Otherwise, it might feel tedious. https://github.com/prometheus/client_golang/pull/1151/#issuecomment-1298340233

Reply:
Does that sound good to you ?